src/extlibs/hrvo/path.cpp

```diff
@@ -0,0 +1,71 @@
+#include "extlibs/hrvo/path.h"
+
+#include <stdexcept>
+#include <utility>
+
+#include "extlibs/hrvo/path_point.h"
+#include "extlibs/hrvo/vector2.h"
+
+
+Path::Path()
+{
+    goal_radius    = 0.0f;
+    Vector2 vector = Vector2();
+    PathPoint path_point(vector);
+    path.push_back(path_point);
+}
+
+Path::Path(const std::vector<PathPoint> &path_points, float goal_radius_)
+{
+    path        = path_points;
+    goal_radius = goal_radius_;
+}
+
+Vector2 Path::getNextPathPointPosition()
+{
+    curr_goal_index++;
+    return getCurrentPathPointPosition();
+}
+
+Vector2 Path::getCurrentPathPointPosition()
+{
+    if (curr_goal_index >= path.size())
+    {
+        return Vector2();
+    }
+    else
+    {
+        return path[curr_goal_index].position_;
+    }
+}
+
+float Path::getDesiredSpeedAtCurrentPathPoint()
+{
+    if (curr_goal_index >= path.size())
+    {
+        return 0.f;
+    }
+    else
+    {
+        return path[curr_goal_index].speed_at_destination;
+    }
+}
+
+bool Path::isGoingToFinalPathPoint()
```

Comment:
instead of this couldn't we return nullopt when next point is called?

Reply:
Note: for the commit recent [4f1a4f6](https://github.com/UBC-Thunderbots/Software/pull/2500/commits/4f1a4f6ec4d0001d12db89c9d42c93ba9d241509), I still kept it as return bool